server/src/main/java/org/opensearch/gateway/GatewayAllocator.java

```diff
@@ -87,15 +99,24 @@ public class GatewayAllocator implements ExistingShardsAllocator {
         ConcurrentCollections.newConcurrentMap();
     private Set<String> lastSeenEphemeralIds = Collections.emptySet();
 
+    private final ConcurrentMap<ShardsBatch, AsyncBatchShardFetch<TransportNodesListGatewayStartedShardsBatch.NodeGatewayStartedShardsBatch>> asyncBatchFetchStarted = ConcurrentCollections.newConcurrentMap();
```

Comment:
Any reason not keeping only batchId in the key ?

It's going to hard to construct whole **ShardsBatch** object just for getting the fetcher object.

Reply:
Keeping batch Id now